### PR TITLE
Upgrade to latest pytorch_pretrained_bert package

### PR DESCRIPTION
**Pull request by iftenney**
New features, and support for some experimental stuff I'm working on.

Needed to start work on #491

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -61,9 +61,9 @@ RUN pip install python-Levenshtein ftfy==5.4.1 spacy==2.0.11
 RUN python -m spacy download en
 
 # Install AllenNLP. Need to update some other deps first.
-RUN pip install pytorch-pretrained-bert==0.4.0
 RUN conda install greenlet=0.4.15
 RUN pip install allennlp==0.8.1
+RUN pip install --upgrade pytorch-pretrained-bert==0.5.1
```
Comment (sleepinyourhat) [inline Dockerfile]: This isn't the latest anymore (as of today).

Comment (iftenney) [reply Dockerfile]: Oh wow, that was quick. Looks like the only changes are related to GPT/GPT2: https://github.com/huggingface/pytorch-pretrained-BERT/compare/v0.5.1...master

**Comment by iftenney** [comment]
This doesn't move GPT, the old version that we're using is a separate package: https://github.com/jsalt18-sentence-repl/jiant/blob/master/.gitmodules#L8

**Comment by iftenney** [comment]
I'm going to merge this as-is since I tested a bunch of runs with 0.5.1. We'll likely want to update again when we start migrating the GPT code.

**Pull request merged**